scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala

```diff
@@ -577,7 +577,8 @@ trait ClassLikeSupport:
 
 
   def unwrapMemberInfo(c: ClassDef, symbol: Symbol): MemberInfo =
-    val baseTypeRepr = typeForClass(c).memberType(symbol)
+    val qualTypeRepr = if c.symbol.isClassDef then This(c.symbol).tpe else typeForClass(c)
```

Comment:
I don't understand this line. Doesn't `This(c.symbol).tpe` return `c.symbol.typeRef`?

Reply:
No, it returns `c.symbol.thisType` (for which we don't have any more direct syntax in the Quotes API, which is why I went this way).

Reply:
I see.